File: backend/strategy-game-common-data/src/main/kotlin/io/github/smiley4/strategygame/backend/commondata/Route.kt
package io.github.smiley4.strategygame.backend.commondata

data class Route(
    val id: Id,
    val worldObjectA: WorldObject.Id,
    val worldObjectB: WorldObject.Id,
    val path: List<Tile.Ref>, // note: includes tiles with start/end world objects
    val cost: Float,
) {

    @JvmInline
    value class Id(val value: String) {
        companion object
    }

}

File: frontend/js/src/models/route/route.ts
import {BrandedId} from "../../common/brandedId";
import {TileSummary} from "../tile/tileSummary";
import {HiddenType} from "../../common/hiddenType";
import {WorldObjectSummary} from "../worldobject/worldObjectSummary";

export interface Route {
    id: Route.Id,
    worldObjectA: HiddenType<WorldObjectSummary>,
    worldObjectB: HiddenType<WorldObjectSummary>,
    cost: number,
    path: TileSummary[]
}


export namespace Route {

    export type Id = BrandedId<string, "RouteId">;

}